Sources/Datadog/Logger.swift

```diff
@@ -398,24 +407,70 @@ public class Logger: LoggerProtocol {
                 )
             }()
 
+            let logger: LoggerProtocol
             switch (remoteLogger, consoleLogger) {
             case (let remoteLogger?, nil):
-                return remoteLogger
+                logger = remoteLogger
 
             case (nil, let consoleLogger?):
-                return consoleLogger
+                logger = consoleLogger
 
             case (let remoteLogger?, let consoleLogger?):
-                return CombinedLogger(combinedLoggers: [remoteLogger, consoleLogger])
+                logger = CombinedLogger(combinedLoggers: [remoteLogger, consoleLogger])
 
             case (nil, nil): // when user explicitly produces a no-op logger
                 return NOPLogger()
             }
+
+            if let samplingRate = samplingRate {
+                return SamplingLogger(logger: logger, sampler: Sampler(samplingRate: samplingRate))
+            } else if let sampler = loggingFeature.configuration.loggingSampler {
+                return SamplingLogger(logger: logger, sampler: sampler)
+            } else {
+                return logger
+            }
         }
     }
 }
 
-/// Combines multiple loggers together into single `Logger` interface.
+/// Wraps a `LoggerProtocol` and enables sampling using the injected instance of `Sampler`.
+internal struct SamplingLogger: LoggerProtocol {
```

Comment:
I don't think you need this wrapper, the sampling should be applied when using the `RemoteLogger`. The `ConsoleLogger` shouldn't be sampled so all logs are printed in the console.

Reply:
Yes, I think we should continue to print everything to the console.

That make me think that we would also need to force `sample rate = 100` when `debugOverride` is true. Like we do with other Features, e,g: https://github.com/DataDog/dd-sdk-ios/blob/develop/Sources/Datadog/Core/FeaturesConfiguration.swift#L217